### PR TITLE
Downgrade placeholder error to warning for swiftparser

### PR DESCRIPTION
**Pull request by kimdv**
https://github.com/apple/swift-syntax/pull/1380

### CODE REVIEW

File: lib/ASTGen/Sources/ASTGen/SourceFile.swift [hunk 1]
```diff
@@ -77,7 +78,8 @@ extension Syntax {
 public func emitParserDiagnostics(
   diagEnginePtr: UnsafeMutablePointer<UInt8>,
   sourceFilePtr: UnsafeMutablePointer<UInt8>,
-  emitOnlyErrors: CInt
+  emitOnlyErrors: CInt,
+  downgradePlaceholderErrors: CInt
```
Comment (ahoppen) [inline lib/ASTGen/Sources/ASTGen/SourceFile.swift:82]: Let’s be even clearer here and call it `downgradePlaceholderErrorsToWarnings`

File: lib/Parse/ParseDecl.cpp [hunk 2]
```diff
@@ -276,7 +277,8 @@ void Parser::parseTopLevelItems(SmallVectorImpl<ASTNode> &items) {
                !Context.Diags.hadAnyError() &&
                swift_ASTGen_emitParserDiagnostics(&Context.Diags,
                                                   SF.exportedSourceFile,
-                                                  /*emitOnlyErrors=*/true)) {
+                                                  /*emitOnlyErrors=*/true,
+                                                  /*downgradePlaceholderErrors=*/true)) {
```
Comment (ahoppen) [inline lib/Parse/ParseDecl.cpp:281]: You should be getting whether to downgrade placeholder errors to warnings from the `LangOpts`. Something like the following

```suggestion
                                                  Context.LangOpts.WarnOnEditorPlaceholder)) {
```

same below

**Comment by ahoppen** [comment]
https://github.com/apple/swift-syntax/pull/1380

@swift-ci Please smoke test

**Comment by kimdv** [comment]
https://github.com/apple/swift-syntax/pull/1380

@swift-ci Please smoke test

File: lib/ASTGen/Sources/ASTGen/SourceFile.swift [hunk 3]
```diff
@@ -94,15 +96,24 @@ public func emitParserDiagnostics(
       if diag.node.isInIfConfig {
         continue
       }
-      if emitOnlyErrors != 0, diag.diagMessage.severity != .error {
+
+      let diagnosticSeverity: DiagnosticSeverity
+      if downgradePlaceholderErrorsToWarnings == 1 && diag.diagMessage.diagnosticID == StaticTokenError.editorPlaceholder.diagnosticID {
+          diagnosticSeverity = .warning
```
Comment (ahoppen) [inline lib/ASTGen/Sources/ASTGen/SourceFile.swift:102]: Indentation should be 2 spaces

**Comment by kimdv** [comment]
https://github.com/apple/swift-syntax/pull/1380

@swift-ci Please smoke test

**Comment by kimdv** [comment]
https://github.com/apple/swift-syntax/pull/1380

@swift-ci Please smoke test

**Review by ahoppen — APPROVED** [review]

**Comment by kimdv** [comment]
https://github.com/apple/swift-syntax/pull/1380

@swift-ci Please smoke test

**Comment by kimdv** [comment]
https://github.com/apple/swift-syntax/pull/1380

@swift-ci Please smoke test

**Comment by kimdv** [comment]
https://github.com/apple/swift-syntax/pull/1380

@swift-ci Please smoke test macOS

**Comment by kimdv** [comment]
https://github.com/apple/swift-syntax/pull/1380

@swift-ci Please smoke test macOS

**Pull request merged**